### PR TITLE
[core] Use util::readFile for Local and Asset file sources

### PR DESCRIPTION
**Pull request by alexshalamov**
Use exception free util::readFile instead of util::read_file for LocalFileSource and AssetFileSource implementations.

Fixes: #9244

### CODE REVIEW

**Comment by alexshalamov** [comment]
@kkaefer `LocalFileSource::Impl::request` and `AssetFileSource::Impl::request` almost identical, should I refactor it in this PR?

File: platform/default/local_file_request.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            response.error = std::make_unique<Response::Error>(
+                Response::Error::Reason::Other,
+                std::string("Cannot read file ") + path);
+        } else {
+            response.data = std::make_shared<std::string>(std::move(*data));
+        }
+    }
+
+    req.invoke(&FileSourceRequest::setResponse, response);
+}
```
Comment (kkaefer) [inline platform/default/local_file_request.hpp]: Why is this implementation in a header file?

File: platform/default/local_file_source.cpp [hunk 2]
```diff
@@ -38,27 +27,8 @@ class LocalFileSource::Impl {
...
-                response.error = std::make_unique<Response::Error>(
-                    Response::Error::Reason::Other,
-                    std::string("Cannot read file ") + path);
-            } else {
-                response.data = std::make_shared<std::string>(std::move(*data));
-            }
-        }
-
-        req.invoke(&FileSourceRequest::setResponse, response);
+        requestLocalFile(path, response, req);
```
Comment (kkaefer) [inline platform/default/local_file_source.cpp]: `std::move(req)`

File: platform/default/local_file_request.hpp (hunk 1 above)
Comment (alexshalamov) [reply platform/default/local_file_request.hpp]: moved to designated cpp file

File: platform/default/local_file_source.cpp (hunk 2 above)
Comment (alexshalamov) [reply platform/default/local_file_source.cpp]: Done.

**Comment by kkaefer** [comment]
bump

**Comment by alexshalamov** [comment]
@kkaefer Sorry, completely missed notification and forgot about this PR. Rebased to master.

**Pull request merged**